soroban-env-host/src/host/declared_size.rs

```diff
@@ -259,7 +260,7 @@ mod test {
         expect!["8"].assert_eq(size_of::<SymbolSmallIter>().to_string().as_str());
         expect!["32"].assert_eq(size_of::<U256>().to_string().as_str());
         expect!["32"].assert_eq(size_of::<I256>().to_string().as_str());
-        expect!["40"].assert_eq(size_of::<HostObject>().to_string().as_str());
+        expect!["48"].assert_eq(size_of::<HostObject>().to_string().as_str());
```

Comment:
@graydon , for me this size is 40 (so I updated it in my previous PR). Is `HostObject` somehow platform-dependent?

Reply:
Certainly could be. Really lots of these can be platform dependent. The way this works is we have an "official" declared size for each type (declared above) that's not-platform-specific, and then this test and the one below checks that the declared size isn't (too) wrong. At the moment we're checking for exact identity, but it can be a little wrong without the sky falling, it's just a source of metering-inaccuracy (of which there are lots).

For this test probably the nicest fix is to confirm why it's happening (platform variability?) and then cfg-guard it by platform, since we're already doing the upper-bound test against declared sizes in the subsequent test. This is really just an eyeball test to catch when a struct suddenly changes size or larger/smaller than expected.

Reply:
I've used this test to figure out the sizes of newly added structs. Interestingly, this seems to be the only discrepancy, which made me wonder if there is something special about host object struct.

Reply:
I dug a bit more into this and found the answer: u128 is 8-byte aligned on x86-64 and 16-byte aligned on aarch64. The {I,U}256 host object types contain two u128 values so are 32 bytes themselves, but since they're stuck in an enum arm they're unaligned by the tag byte and so to align them again winds up chewing up 32+16=48 bytes total.